source/module_io/json_output/test/para_json_test.cpp

```diff
@@ -10,51 +10,112 @@ TEST(AbacusJsonTest, AddJson)
     Json::AbacusJson::doc.SetObject();
 
     // add a string
-    Json::AbacusJson::add_json({"key1"}, "value1");
+    Json::AbacusJson::add_json({"key1"}, "value1",false);
     ASSERT_TRUE(Json::AbacusJson::doc.HasMember("key1"));
     ASSERT_TRUE(Json::AbacusJson::doc["key1"].IsString());
     ASSERT_STREQ(Json::AbacusJson::doc["key1"].GetString(), "value1");
 
     // add a string to a nested object
-    Json::AbacusJson::add_json({"key2", "key3"}, "value2");
+    Json::AbacusJson::add_json({"key2", "key3"}, "value2",false);
     ASSERT_TRUE(Json::AbacusJson::doc.HasMember("key2"));
     ASSERT_TRUE(Json::AbacusJson::doc["key2"].IsObject());
     ASSERT_TRUE(Json::AbacusJson::doc["key2"].HasMember("key3"));
     ASSERT_TRUE(Json::AbacusJson::doc["key2"]["key3"].IsString());
     ASSERT_STREQ(Json::AbacusJson::doc["key2"]["key3"].GetString(), "value2");
 
     // add an int
-    Json::AbacusJson::add_json({"key2"}, 123);
+    Json::AbacusJson::add_json({"key2"}, 123,false);
     ASSERT_TRUE(Json::AbacusJson::doc.HasMember("key2"));
     ASSERT_TRUE(Json::AbacusJson::doc["key2"].IsInt());
     ASSERT_EQ(Json::AbacusJson::doc["key2"].GetInt(), 123);
 
     // add a bool
-    Json::AbacusJson::add_json({"key3"}, true);
+    Json::AbacusJson::add_json({"key3"}, true,false);
     ASSERT_TRUE(Json::AbacusJson::doc.HasMember("key3"));
     ASSERT_TRUE(Json::AbacusJson::doc["key3"].IsBool());
     ASSERT_EQ(Json::AbacusJson::doc["key3"].GetBool(), true);
 
     // add a double
-    Json::AbacusJson::add_json({"key4"}, 1.23);
+    Json::AbacusJson::add_json({"key4"}, 1.23,false);
     ASSERT_TRUE(Json::AbacusJson::doc.HasMember("key4"));
     ASSERT_TRUE(Json::AbacusJson::doc["key4"].IsDouble());
     ASSERT_EQ(Json::AbacusJson::doc["key4"].GetDouble(), 1.23);
 
     // modify a value
-    Json::AbacusJson::add_json({"key4"}, 4.56);
+    Json::AbacusJson::add_json({"key4"}, 4.56,false);
     ASSERT_EQ(Json::AbacusJson::doc["key4"].GetDouble(), 4.56);
+
+    // modify a value
+    Json::AbacusJson::add_json({"key4"}, 4.56,false);    
+
+
+    //array test
+    Json::AbacusJson::add_json({"key6","key7"}, true,true);
+    Json::AbacusJson::add_json({"key6","key7"}, false,true);
+
+    
+    // add key-val to a object array
+    for(int i=0;i<3;i++){
+        Json::jsonValue object(JobjectType);
+        object.JaddNormal("int",i);
+        std::string str = std::to_string(i*100);   
+        object.JaddString("string", str);
+        object.JaddNormal("double", 0.01*i);    
```

Comment:
suggest adding a function to package the action to create a json object.

Reply:
By design, this is the easiest way to construct an object. Because object is required to add a key-val pair, and if you use the wrap function, you also need to construct a separate key-value pair argument list. It won't be any simpler than it is now.